### ISSUE TITLE
Traceback (most recent call last):   File "generate_tfrecord.py", line 110, in <module>     tf.app.run() AttributeError: module 'tensorflow' has no attribute 'app'

### ISSUE THREAD

**Issue by hemanth358** [EdjeElectronics/TensorFlow-Object-Detection-API-Tutorial-Train-Multiple-Objects-Windows-10#371]
I also changed none to return 0 in else block in generate_tfrecord.py
but not able to figureout the issue

**Comment by YumiAlmero03** [comment]
hi i encounter the same

**Comment by YumiAlmero03** [comment]
i encounter this : https://www.tensorflow.org/guide/upgrade it changes 'tf.app.run' to 'tf.compat.v1.app.run' but there are still problem with flags because it is also deprecated

it quoted 
"Because of TensorFlow 2.x module deprecations (for example, tf.flags and tf.contrib), some changes can not be worked around by switching to compat.v1. Upgrading this code may require using an additional library (for example, absl.flags) or switching to a package in tensorflow/addons."

so i hope we need is to fix the flags

**Comment by kashifkhankk** [comment]
Stuck on this error  AttributeError: module 'tensorflow' has no attribute 'app' unable to find any workaround

**Comment by alihafidz** [comment]
@kashifkhankk just replace your tf.app.run() with this tf.compat.v1.app.run()

**Comment by YumiAlmero03** [comment]
@alihafidz yes but  i still have errors in the flags it doesn't read the content

**Comment by hemanth358** [comment]
> 
> 
> i encounter this : https://www.tensorflow.org/guide/upgrade it changes 'tf.app.run' to 'tf.compat.v1.app.run' but there are still problem with flags because it is also deprecated
> 
> it quoted
> "Because of TensorFlow 2.x module deprecations (for example, tf.flags and tf.contrib), some changes can not be worked around by switching to compat.v1. Upgrading this code may require using an additional library (for example, absl.flags) or switching to a package in tensorflow/addons."
> 
> so i hope we need is to fix the flags

is that working after renaming flags to  absl.flags

**Comment by kashifkhankk** [comment]
@alihafidz  Thanks it is working i modified generate_tfrecord.py, on line no 17  (tensorflow to  tensorflow.compat.v1 )

now stuck on module 'tensorflow' has no attribute 'contrib'

**Comment by JulianOrteil** [comment]
@kashifkhankk 
You have Tensorflow 2.0 installed. You need Tensorflow 1.x. The object detection API has not yet been updated to run with Tensorflow 2.0 (running the upgrade script against the API will not work either, just use TF 1.x).

Revert any changes you made to generate_tfrecord.py or any other file that you may have added "v1" to then run this command (if you are using conda, run it in the conda environment):

**pip install tensorflow-gpu==1.14.0** (for GPU version)
**pip install tensorflow==1.14.0** (for CPU version)

TF 2.0 does not contain the 'contrib' module has it has been integrated into the main distribution of TF 2.0 so you cannot fix that unless if you want to rewrite thousands of lines of code within Tensorflow itself.

This should also fix your original 'app' issue. TF 2.0 does not contain that module either.
